### ISSUE TITLE
Fails to parse async lambdas with default parameters

### ISSUE THREAD

**Issue by kamek-pf** [babel/babel-eslint#379]
This is valid : 
```javascript
const myFunc1 = async function (num: number, str: string = 'defaultStrVal'): Promise<any> {
    return await new Promise((resolve) => resolve(str));
};

const myFunc2 = async (num: number, str: string): Promise<any> => {
    return await new Promise((resolve) => resolve(str));
};
```

However, this fails to parse with `Unexpected token` : 
```javascript
const myFunc3 = async (num: number, str: string = 'defaultStrVal'): Promise<any> => {
    return await new Promise((resolve) => resolve(str));
};
```

Relevant parts of my package.json :
```json
...
    "babel-cli": "^6.6.5",
    "babel-eslint": "^6.0.5",
    "babel-plugin-transform-flow-strip-types": "^6.14.0",
    "eslint": "^3.4.0",
    "eslint-plugin-babel": "^3.3.0",
    "flow-bin": "^0.31.1"
...
```

eslintrc has `"parser": "babel-eslint"`.

Any idea what's causing this ?

**Comment by kaicataldo** [comment]
This parses for me in the Babel REPL. Mind sharing your .eslintrc config?

**Comment by kamek-pf** [comment]
@kaicataldo 
Well that's weird, I just tried again in my own code and I can't reproduce anymore. The REPL also works, no idea what happened.

Anyway, sorry for the noise, closing this.

**Comment by kaicataldo** [comment]
Glad it works 😄
